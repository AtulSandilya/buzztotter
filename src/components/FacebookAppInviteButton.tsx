import * as React from "react";
import { Component} from 'react';
import {Text, TouchableOpacity, View, } from 'react-native';

import { AppInviteDialog } from 'react-native-fbsdk';

import Icon from 'react-native-vector-icons/Ionicons';

const FacebookAppInviteButton: React.StatelessComponent<{}> = () => {
  // TODO: Link this to the actual website
  const appLinkUrl = "https://fb.me/2145387155685498";

  return (
    <View
      style={{
        flex: -1,
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: 15,
      }}
    >
      <Icon.Button
        name="logo-facebook"
        backgroundColor="#3b5988"
        onPress={() => {
          AppInviteDialog.show({
            applinkUrl: appLinkUrl,
          })
          .then((result) => {
            return true;
          })
        }}
      >
        Invite Friends
      </Icon.Button>
    </View>
  )
}

export default FacebookAppInviteButton;
