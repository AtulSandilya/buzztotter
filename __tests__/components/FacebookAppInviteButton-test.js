import 'react-native';
import React from 'react';
import FacebookAppInviteButton from '../../build/components/FacebookAppInviteButton';

import renderer from 'react-test-renderer';

describe('Facebook App Invite Button', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <FacebookAppInviteButton />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
