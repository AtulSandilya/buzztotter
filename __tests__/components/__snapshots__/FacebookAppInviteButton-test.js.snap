exports[`Facebook App Invite Button renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": -1,
      "justifyContent": "center",
      "marginTop": 15,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#3b5988",
            "borderRadius": 5,
          },
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "backgroundColor": "#3b5988",
            "borderRadius": 5,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "marginRight": 10,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        ïˆ¶
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontWeight": "600",
            },
            Object {
              "color": "white",
            },
          ]
        }>
        Invite Friends
      </Text>
    </View>
  </View>
</View>
`;
